core/src/test/scala/org/apache/spark/scheduler/TaskSetManagerSuite.scala

```diff
@@ -1139,6 +1138,19 @@ class TaskSetManagerSuite extends SparkFunSuite with LocalSparkContext with Logg
       .updateBlacklistForFailedTask(anyString(), anyString(), anyInt())
   }
 
+  test("Schedule tasks based on size of input from ShuffledRDD.") {
+    sc = new SparkContext("local", "test")
+    sched = new FakeTaskScheduler(sc)
+    val taskSet = FakeTask.createTaskSet(4)
+    val clock = new ManualClock()
+    val manager = new TaskSetManager(sched, taskSet, 1, clock = clock)
+    manager.setTaskInputSizeFromShuffledRDD(taskSet.tasks.zip(Seq(1L, 100L, 10000L, 1000L)).toMap)
+    assert(manager.resourceOffer("exec", "host", ANY).get.index === 2)
+    assert(manager.resourceOffer("exec", "host", ANY).get.index === 3)
+    assert(manager.resourceOffer("exec", "host", ANY).get.index === 1)
+    assert(manager.resourceOffer("exec", "host", ANY).get.index === 0)
+  }
```

Comment:
we'd also want a test to make sure the sizes were getting computed correctly.  (I think that might be easier to do with the refactor I suggested?)